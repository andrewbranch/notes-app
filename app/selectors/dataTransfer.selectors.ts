import { StateShape } from '../reducers';
import { createSelector } from 'reselect';

export const dataTranferSelector = (state: StateShape) => state.dataTransfer;
export const loadedNotesStatusSelector = createSelector(dataTranferSelector, dataTransfer => dataTransfer.loadedNotesStatus);