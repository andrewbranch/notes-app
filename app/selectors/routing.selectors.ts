import { createSelector } from 'reselect';
import { StateShape } from '../reducers';

export const routingSelector = (state: StateShape) => state.routing;
export const locationSelector = createSelector(routingSelector, routing => routing.location);
export const selectedNoteIdSelector = createSelector(
  locationSelector,
  (state: StateShape) => state.notes,
  (location, notes) => {
    const noteId = location!.pathname.substring(1);
    return Object.keys(notes).includes(noteId) ? noteId : null;
  }
);
