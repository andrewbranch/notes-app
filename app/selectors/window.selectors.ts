import { createSelector } from 'reselect';
import { StateShape } from '../reducers';

export const windowSelector = (state: StateShape) => state.window;
export const showEditorDebuggerSelector = createSelector(windowSelector, window => window.showEditorDebugger);