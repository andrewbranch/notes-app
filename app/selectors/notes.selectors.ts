import * as memoize from 'memoizee';
import createCachedSelector from 're-reselect';
import { sortBy } from 'lodash';
import { createSelector } from 'reselect';
import { EditorState, convertFromRaw } from 'draft-js';
import { StateShape } from '../reducers';
import { selectedNoteIdSelector } from './routing.selectors';
import { NotesState, Note, RawNote } from '../reducers/types';
import { getNoteTitle } from '../utils/getNoteTitle';

export const notesSelector = (state: StateShape) => state.notes;
export const noteIdsSelector = createSelector(notesSelector, notes => Object.keys(notes));
export const sortedNoteIdsSelector = createSelector(
  notesSelector,
  noteIdsSelector,
  (notes, noteIds) => sortBy(noteIds, id => -notes[id].updatedAt)
);

export const selectedRawNoteSelector = createSelector(
  notesSelector,
  selectedNoteIdSelector,
  (notes, selectedNoteId) => (
    selectedNoteId ? notes[selectedNoteId] : null
  )
);

export const rawNoteSelector = createCachedSelector(
  notesSelector,
  (state: StateShape, noteId: string) => noteId,
  (notes, noteId): RawNote => notes[noteId]
)(
  (state, noteId) => noteId
);

export const noteTitleSelector = createCachedSelector(
  rawNoteSelector,
  note => getNoteTitle(note.editor ? note.editor.getCurrentContent() : note.content)
)(
  (state, noteId) => noteId
);

const noteEditorSelector = createCachedSelector(
  rawNoteSelector,
  note => (
    note.editor || EditorState.createWithContent(convertFromRaw(note.content))
  )
)(
  (state, noteId) => noteId
);

export const noteSelector = createCachedSelector(
  rawNoteSelector,
  noteTitleSelector,
  noteEditorSelector,
  (note, title, editor): Note => ({ ...note, title, editor })
)(
  (state, noteId) => noteId
);

export const selectedNoteSelector = createSelector(
  (state: StateShape) => state,
  noteIdsSelector,
  selectedNoteIdSelector,
  (state, noteIds, selectedNoteId) => noteIds.length && selectedNoteId ? noteSelector(state, selectedNoteId) : null
);
