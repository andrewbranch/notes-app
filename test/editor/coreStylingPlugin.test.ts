import { Selector } from './selectors';
import { getContentState, loadApp, pressKey, getState } from './transport';

beforeEach(async () => {
  await loadApp();
});

describe('coreStylingPlugin', () => {
  describe('steps', () => {
    describe('removeInlineStyles', () => {
      describe('deleting decorator characters', () => {
        test('backspacing the first of a two-character leading decorator sequence should remove the style', async () => {
          await page.type(Selector.ContentEditable, '**Bold**');
          await pressKey('ArrowLeft', 7);
          await pressKey('Backspace');
          expect(await getState()).toMatchSnapshot();
        });
      });
    });
  });
});
