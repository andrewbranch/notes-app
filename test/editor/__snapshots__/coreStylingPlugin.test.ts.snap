// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coreStylingPlugin steps removeInlineStyles deleting decorator characters backspacing the first of a two-character leading decorator sequence should remove the style 1`] = `
Object {
  "content": Immutable.Record {
    "entityMap": Object {},
    "blockMap": Immutable.OrderedMap {
      "0": Immutable.Record {
        "key": "0",
        "type": "unstyled",
        "text": "*Bold**",
        "characterList": Immutable.List [
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
          Immutable.Record {
            "style": Immutable.OrderedSet [],
            "entity": null,
          },
        ],
        "depth": 0,
        "data": Immutable.Map {},
      },
    },
    "selectionBefore": Immutable.Record {
      "anchorKey": "0",
      "anchorOffset": 0,
      "focusKey": "0",
      "focusOffset": 0,
      "isBackward": false,
      "hasFocus": false,
    },
    "selectionAfter": Immutable.Record {
      "anchorKey": "0",
      "anchorOffset": 0,
      "focusKey": "0",
      "focusOffset": 0,
      "isBackward": false,
      "hasFocus": false,
    },
  },
  "selection": Immutable.Record {
    "anchorKey": "",
    "anchorOffset": 0,
    "focusKey": "",
    "focusOffset": 0,
    "isBackward": false,
    "hasFocus": false,
  },
}
`;
